### DISCUSSION TITLE
Changing workspaces?

### DISCUSSION THREAD

**Discussion by Sahasrara** [TypeFox/monaco-languageclient#569] [Q&A] [answered]
I see this parameter you can provide to a new instance of MonacoLanguageClient:
`MonacoLanguageClientOptions.LanguageClientOptions.workspaceFolder`

I'm assuming that will tell the language server to go scan that folder/s for code to use in autocomplete suggestions.

My questions are:
1. How can I change the workspace folders after creating the MonacoLanguageClient?
2. How can you monitor the progress of workspace indexing?

**Comment by msujew** [comment, marked as answer]
Hey @Sahasrara 

> I'm assuming that will tell the language server to go scan that folder/s for code to use in autocomplete suggestions.

Depends on the language server, but that's usually correct.

> How can I change the workspace folders after creating the MonacoLanguageClient?

It depends on the language server you're using:

1. If it supports the [`workspace/didChangeWorkspaceFolders`](https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWorkspaceFolders) request, you should just be able to send that one.
2. Otherwise, you'll need to probably restart the language server/client and instruct it to use the correct workspace folder.

> How can you monitor the progress of workspace indexing?

There is no LSP method for doing that. If you have direct control over the language server, you can implement a custom notification to the language client for that (and listen for it on the language client). If you don't, there is basically no way to tell from the outside whether the language server is still working on indexing the workspace.
